engine/src/main/java/org/terasology/logic/console/commands/CoreCommands.java

```diff
@@ -253,10 +249,29 @@ public String search(@CommandParam("searched") String searched) {
      * @return List of blocks that match searched string
      */
     private List<String> findBlockMatches(String searchLowercase) {
-        return assetManager.getAvailableAssets(BlockFamilyDefinition.class)
-                .stream().<Optional<BlockFamilyDefinition>>map(urn -> assetManager.getAsset(urn, BlockFamilyDefinition.class))
-                .filter(def -> def.isPresent() && def.get().isLoadable() && matchesSearch(searchLowercase, def.get()))
-                .map(r -> new BlockUri(r.get().getUrn()).toString()).collect(Collectors.toList());
+        ResourceUrn curUrn;
+        List<String> l=new ArrayList<String>();
+        for (ResourceUrn urn : assetManager.getAvailableAssets(BlockFamilyDefinition.class)) {
+            // Current urn for logging purposes to find the broken urn
+            curUrn = urn;
+            try{
+                Optional<BlockFamilyDefinition> def = assetManager.getAsset(urn, BlockFamilyDefinition.class);
+                if (def.isPresent() && def.get().isLoadable() && matchesSearch(searchLowercase, def.get())) {
+                    String s = new BlockUri(def.get().getUrn()).toString();
+                    l.add(s);
+                }
+            }
+            // If a prefab is broken , it will throw an exception
+            catch(Exception e){
+                List<String> EmptyList=new ArrayList<String>();
+                console.addMessage("Note : Search may not return results if invalid assets are present");
+                console.addMessage("Error parsing : "+curUrn.toString());
+                console.addMessage(e.toString());
+                return(EmptyList);
```

Comment:
Would it be possible to instead of returning here, to just keep looping through the `for` block? 
That way the broken assets will be skipped but the rest of the command will still work and the printing will still work.

Reply:
Good idea, Sorry for not implementing this initially , it was on my mind but I forgot about it once I was done with the functionality . This is the new output after this change.
![Screenshot from 2020-03-20 19-26-20](https://user-images.githubusercontent.com/32770414/77170830-9ea4ae80-6ae1-11ea-8d6d-589f04f44ea4.png)